### DISCUSSION TITLE
Conversation Details API does not return all messages

### DISCUSSION THREAD

**Discussion by ptrln** [chatwoot/chatwoot#10166] [Chatwoot APIs] [answered]
The Conversation Details API does not return all messages in the conversation, despite what the documentation says. It seems to always return only a single message, and there seems to be no way to retrieve all messages in the conversation. Is this a bug?

https://www.chatwoot.com/developers/api/#tag/Conversations/operation/get-details-of-a-conversation

**Comment by sojan-official** [comment, marked as answer]
You need to use the [Messages API](https://www.chatwoot.com/developers/api/#tag/Messages/operation/list-all-messages) to retrieve all messages in a conversation.

Additionally, pagination is required as not all messages are returned in a single call. You can inspect the browser’s console under the Network tab to monitor the corresponding API requests.
